<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SortingNetworkMakerClass</class>
 <widget class="QMainWindow" name="SortingNetworkMakerClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>720</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Sorting Network Maker</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../rsrc/sorting_network_maker.qrc">
    <normaloff>:/icon/icon.png</normaloff>:/icon/icon.png</iconset>
  </property>
  <property name="locale">
   <locale language="English" country="UnitedStates"/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QGridLayout" name="gridLayout" columnstretch="3,1" columnminimumwidth="0,0">
      <property name="leftMargin">
       <number>32</number>
      </property>
      <property name="rightMargin">
       <number>32</number>
      </property>
      <property name="horizontalSpacing">
       <number>32</number>
      </property>
      <item row="0" column="0">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <property name="spacing">
         <number>32</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QComboBox" name="selectAlgorithm">
          <property name="minimumSize">
           <size>
            <width>256</width>
            <height>0</height>
           </size>
          </property>
          <property name="maxVisibleItems">
           <number>20</number>
          </property>
          <item>
           <property name="text">
            <string>Bubble sort</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Odd-even transposition sort</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Comb sort (3-smooth)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Bitonic merge sort (top-down)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Bitonic merge sort (bottom-up)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Odd-even merge sort (top-down)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Odd-even merge sort (bottom-up)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Merge-exchange sort</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Pairwise sort</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Bose Nelson merge sort</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Hibbard merge sort</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Optimal size</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Optimal latency</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="sizeLabel">
          <property name="minimumSize">
           <size>
            <width>128</width>
            <height>0</height>
           </size>
          </property>
          <property name="text">
           <string>Input size:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="selectSize">
          <property name="minimumSize">
           <size>
            <width>128</width>
            <height>0</height>
           </size>
          </property>
          <property name="layoutDirection">
           <enum>Qt::LeftToRight</enum>
          </property>
          <property name="maximum">
           <number>128</number>
          </property>
          <property name="value">
           <number>8</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="0" column="1">
       <widget class="QPushButton" name="buttonGenerate">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>128</width>
          <height>0</height>
         </size>
        </property>
        <property name="whatsThis">
         <string>Generate the picture representing the sorting network. (Shortcut: Enter)</string>
        </property>
        <property name="text">
         <string>generate</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QPushButton" name="buttonSave">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="minimumSize">
         <size>
          <width>128</width>
          <height>0</height>
         </size>
        </property>
        <property name="whatsThis">
         <string>Save the generated picture. (Shortcut: Ctrl+S)</string>
        </property>
        <property name="text">
         <string>save</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <property name="spacing">
         <number>32</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QLabel" name="opNameLabel">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="whatsThis">
           <string>The number of compare-and-swap operations in a sorting network.</string>
          </property>
          <property name="text">
           <string>Number of Comparators:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="opValueLabel">
          <property name="whatsThis">
           <string>The number of compare-and-swap operations in a sorting network.</string>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="latencyNameLabel">
          <property name="whatsThis">
           <string>When a sorting network can be executed with arbitrary parallelism, here shows its latency.</string>
          </property>
          <property name="text">
           <string>Latency:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="latencyValueLabel">
          <property name="whatsThis">
           <string>When a sorting network can be executed with arbitrary parallelism, here shows its latency.</string>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QScrollArea" name="scrollArea">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>1</horstretch>
        <verstretch>1</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>256</height>
       </size>
      </property>
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>1256</width>
         <height>559</height>
        </rect>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QLabel" name="showPicture">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>1</horstretch>
            <verstretch>1</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="scaledContents">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1280</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionGenerate"/>
    <addaction name="actionSave"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionWhatsThis"/>
    <addaction name="separator"/>
    <addaction name="actionAbout"/>
    <addaction name="actionQt"/>
   </widget>
   <widget class="QMenu" name="menuSettings">
    <property name="title">
     <string>Settings</string>
    </property>
    <widget class="QMenu" name="menuTestExample">
     <property name="title">
      <string>Test Example</string>
     </property>
     <addaction name="actionShowTestExample"/>
     <addaction name="actionExampleFont"/>
     <addaction name="separator"/>
     <addaction name="actionTestReproducible"/>
     <addaction name="actionTestStability"/>
    </widget>
    <widget class="QMenu" name="menuColor">
     <property name="title">
      <string>Color</string>
     </property>
     <addaction name="actionLineColor"/>
     <addaction name="actionBackgroundColor"/>
     <addaction name="actionBackgroundTransparency"/>
    </widget>
    <widget class="QMenu" name="menuLayout">
     <property name="title">
      <string>Layout</string>
     </property>
     <addaction name="actionSplitParallel"/>
     <addaction name="actionSplitRecursive"/>
     <addaction name="actionCompact"/>
    </widget>
    <widget class="QMenu" name="menuResolution">
     <property name="title">
      <string>Resolution</string>
     </property>
     <addaction name="actionResolution"/>
     <addaction name="actionKeepAspectRatio"/>
    </widget>
    <addaction name="menuColor"/>
    <addaction name="menuResolution"/>
    <addaction name="menuLayout"/>
    <addaction name="menuTestExample"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuSettings"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionSave">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="whatsThis">
    <string>Save the generated picture.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionQt">
   <property name="text">
    <string>About Qt</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About Sorting Network Maker</string>
   </property>
  </action>
  <action name="actionLineColor">
   <property name="text">
    <string>Line Color</string>
   </property>
  </action>
  <action name="actionBackgroundColor">
   <property name="text">
    <string>Background Color</string>
   </property>
  </action>
  <action name="actionBackgroundTransparency">
   <property name="text">
    <string>Background Transparency</string>
   </property>
  </action>
  <action name="actionKeepAspectRatio">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Keep Aspect Ratio</string>
   </property>
  </action>
  <action name="actionResolution">
   <property name="text">
    <string>Set Resolution</string>
   </property>
  </action>
  <action name="actionShowTestExample">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show Test Example</string>
   </property>
  </action>
  <action name="actionTestReproducible">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Reproducible Randomness</string>
   </property>
   <property name="whatsThis">
    <string>When the option &quot;Reproducible Randomness&quot; is enabled, the generated test example will be the same in each repeated generation.</string>
   </property>
  </action>
  <action name="actionExampleFont">
   <property name="text">
    <string>Font</string>
   </property>
  </action>
  <action name="actionTestStability">
   <property name="text">
    <string>Test Sorting Stability</string>
   </property>
   <property name="whatsThis">
    <string>A sorting algorithm is defined to be stable if it preserves the relative order of equal elements. Here configures the test examples to test the stability of the sorting networks.</string>
   </property>
  </action>
  <action name="actionWhatsThis">
   <property name="text">
    <string>What's this?</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
  <action name="actionCompact">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Compact Layout</string>
   </property>
   <property name="whatsThis">
    <string>When the option &quot;Compact Layout&quot; is enabled, the comparators may be reordered to make the layout more compact.</string>
   </property>
  </action>
  <action name="actionSplitRecursive">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Split Recursive Levels</string>
   </property>
   <property name="whatsThis">
    <string>The option &quot;Split Recursive Levels&quot; is only effective when the option &quot;Split Parallel Levels&quot; is enabled. When the option &quot;Split Recursive Levels&quot; is enabled, the comparators in recursive algorithms cannot be reordered across recursive levels.</string>
   </property>
  </action>
  <action name="actionSplitParallel">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Split Parallel Levels</string>
   </property>
   <property name="whatsThis">
    <string>When the option &quot;Split Parallel Levels&quot; is enabled, the comparators will be sorted by their parallel latency. Also, comparators different latency cannot be drawn in the same column.</string>
   </property>
  </action>
  <action name="actionGenerate">
   <property name="text">
    <string>Generate</string>
   </property>
   <property name="shortcut">
    <string>Enter</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="../rsrc/sorting_network_maker.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>SortingNetworkMakerClass</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>541</x>
     <y>422</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSplitParallel</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionSplitRecursive</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionShowTestExample</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionExampleFont</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
